libbeat/outputs/shipper/shipper.go

```diff
@@ -43,14 +43,29 @@ import (
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
+type pendingBatch struct {
+	batch        publisher.Batch
+	index        uint64
+	serverID     string
```

Comment:
It's needed to handle the edge case when we are connected to a server that didn't accept the batch. In this case we must consider it lost and cancel.